pkg/stanza/operator/input/windows/input.go

```diff
@@ -55,15 +101,25 @@ func (i *Input) Start(persister operator.Persister) error {
 		}
 	}
 
-	i.subscription = NewSubscription()
-	if err := i.subscription.Open(i.channel, i.startAt, i.bookmark); err != nil {
-		return fmt.Errorf("failed to open subscription: %w", err)
-	}
-
 	i.publisherCache = newPublisherCache()
 
+	var subscription Subscription
+	if i.isRemote() {
+		subscription = NewRemoteSubscription(i.remote.Server)
+		if err := subscription.Open(i.channel, i.startAt, i.bookmark, uintptr(i.remoteSessionHandle)); err != nil {
+			i.Logger().Error("Failed to open subscription for remote server", zap.String("server", i.remote.Server), zap.Error(err))
+		}
+	} else {
+		subscription = NewLocalSubscription()
+		if err := subscription.Open(i.channel, i.startAt, i.bookmark, 0); err != nil {
+			i.Logger().Error("Failed to open local subscription", zap.Error(err))
```

Comment:
Not exactly sure where a separate PR should start in this case since a lot of the code was not tested beforehand for the input. I added some test cases for on start up though. 

Now if a bad channel name used, it will produce an error and will fail to start.
```log
Error: cannot start pipelines: start stanza: failed to open subscription for remote server: failed to subscribe to bad-channel channel: The specified channel could not be found.
2024/07/15 15:10:58 collector server run finished with error: cannot start pipelines: start stanza: failed to open subscription for remote server: failed to subscribe to bad-channel channel: The specified channel could not be found.
```

Reply:
What I meant is that the behavior should have tests so it is not changed by accident. As you said such test doesn't exist that's why I mentioned it as a possible separated PR.